posts: Add 'GSoC23 Final Report' post

davidbtadokoro opened the pull request.

rodrigosiqueira commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,195 @@
+tags: [gsoc23, kw, kw patch-hub, lore, linux]
+---
+
+My GSoC23 journey, which I introduced in a [previous post](), is almost over. It really doesn't feel like 16 weeks have passed, but I can say that, in this period, I have learned a lot and grown as a developer.
```
I guess you missed the link in the `previous post` part.

rodrigosiqueira commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,195 @@
+
+---
+
+This first section is to describe my contributions to kw that are not directly related to my GSoC23 project. Nonetheless, these were important in their own context and made me more in sync with kw coding style, its contribution model, and, most importantly, with my mentors and people around the project (which I found most invaluable).
```
How about: This first section describes my contributions to kw that are not directly related to the patch-hub feature but were part of my GSoC.

rodrigosiqueira commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,195 @@
+
+### **Adding Support for Native Zsh Completions**
+
+Before GSoC23, I already was contributing to kw and the first big one (both in scope and in number of commits) that I did was adding support for native Zsh completions. Without getting into too much detail, each Shell, say, Bash, can provide command completions. In other words, it's the well-known behavior of hitting `TAB` and waiting for the Shell to either complete the command you are typing or ...
```
>  I already was contributing to kw and the first big one (both in scope and in number of commits) that I did was adding support for native Zsh completions.

You made those contributions in the context of the GSoC application. How about something like this:

> I contributed meaningful changes to kw project during the application period. My first significant contribution (both in scope and number of commits) was adding support for native Zsh completions.

rodrigosiqueira commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,195 @@
+
+Before GSoC23, I already was contributing to kw and the first big one (both in scope and in number of commits) that I did was adding support for native Zsh completions. Without getting into too much detail, each Shell, say, Bash, can provide command completions. In other words, it's the well-known behavior of hitting `TAB` and waiting for the Shell to either complete the command you are typing or ...
+
+These completions are Shell-dependent, and kw only had native support for Bash completions. Zsh completions were adapted from the native Bash completions, but this "emulation" didn't work and resulted in broken completions for Zsh. This was a waste, as the Zsh completion system (although really complicated to work on) provided deeper features than Bash, like highlighting options shown, coupling d ...
```
`although really complicated to work on` drop this part.

rodrigosiqueira commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,195 @@
+
+These completions are Shell-dependent, and kw only had native support for Bash completions. Zsh completions were adapted from the native Bash completions, but this "emulation" didn't work and resulted in broken completions for Zsh. This was a waste, as the Zsh completion system (although really complicated to work on) provided deeper features than Bash, like highlighting options shown, coupling d ...
+
+During February I worked on bringing kw support for native Zsh completions. I described this in further detail in an [earlier post](), but you can see the full Pull Request by clicking [here](https://github.com/kworkflow/kworkflow/pull/773). To illustrate, below is a demo of the results achieved.
```
Add post link.

rodrigosiqueira commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,195 @@
+
+From before the Community Bonding Period began until halfway through it (from mid-April to mid-May), I worked on introducing the Database Management System (DBMS) SQLite3 to kw. This was a long-awaited addition for the kw community, as it would improve the project's scalability and allow the collection of statistics.
+
+I discussed this in further detail in an [earlier post]() and the full Pull Request representing this contribution can be accessed [here](https://github.com/kworkflow/kworkflow/pull/836).
```
add link

rodrigosiqueira commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,195 @@
+
+Since my proposal, my better understanding of the problem at hand along with my interactions with my mentors, made me realize that the most important deliverable was to provide a reliable UI to lore.
+
+Not that we planned on having discrete development cycles, but looking in hindsight, I can divide my work on `kw patch-hub` into 3 development cycles. The first cycle was related to experimenting and understanding the problem and building the feature's core.
```
Do you mean `Note`?

rodrigosiqueira commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,195 @@
+
+Not that we planned on having discrete development cycles, but looking in hindsight, I can divide my work on `kw patch-hub` into 3 development cycles. The first cycle was related to experimenting and understanding the problem and building the feature's core.
+
+From an organizational perspective, we documented every starting requisite in issues and added them to a GitHub Kanbam board. I wish I could provide a link to this board, but its visibility is private, so below is a print of it, just for illustration purposes.
```
I didn't know the kanbam was private; sorry for that. I made It public.

rodrigosiqueira commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,195 @@
+
+![kw patch-hub First Cycle]({{site.url}}/images/gifs/kw_patch_hub_first_cycle.gif)
+
+From mid-May up until mid-June, those were my contributions in the form of Pull Requests, in chronological order of merge:
```
Maybe highlight the number of commits per PR.

rodrigosiqueira commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,195 @@
+
+##### Refactoring the Controller
+
+I described in a [previous post]() the Finite-State Machine computation model used to implement `kw patch-hub` Controller, but the thing was that for each new state added, `src/upstream_patches_ui.sh` grew uncontrollably. At one moment, the file was more than 500 lines in size with functions that didn't follow a logical order, which made it harder and harder to scroll to the desired line each tim ...
```
Add link

rodrigosiqueira commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,195 @@
+7. Allow users to reply patchsets with 'Reviewed-by', 'Tested-by', and with inline reviews.
+8. Improve feature UX.
+9. Refine feature fixing bugs.
+
```
Let everyone know that you will keep working on the project until the end of this year since this is also your capstone project.

rodrigosiqueira reviewed (changes requested):
Please keep the lines short to make it easy to review the new version.

davidbtadokoro commented:
Thank you so much for the review @rodrigosiqueira.

Sorry for the delay. I updated the post on my site yesterday, but forgot to update the PR here. The links to the posts that you commented were left blank because these posts aren't already in kw's blog. I added links to the posts on my site and will change them to point to kw's blog if they get merged into it.

Thanks again for the review!

melissawen commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,344 @@
+
+---
+
+Considering the deliverables, I implemented number 1 and part of number 2. More
```
I didn't get what number 1 and 2 are here... what are the deliverables? Also, I really think you've delivered a huge work, but the way you expressed it here doesn't make it clear... it just the tone, but I'd prefer something like:

```
[As a result of my GSoC project, `kw patch-hub` can be used as a reliable] UI to the lore archives
and provides some other functionalities like bookmarking patchsets, downloading
applicable patchsets (to a default or custom directory), and managing the feature's
settings through the feature itself.
```

davidbtadokoro replied:
You are right. That's a much better way to say it. Also, thanks a lot for the compliment in my work :)

melissawen commented on _posts/2023-08-26-gsoc23-final-report.md:
```diff
@@ -0,0 +1,344 @@
+further than that.
+
+Here is a list of practically all steps needed for `kw patch-hub` to be considered
+"complete", not necessarily in the order of importance:
```
I'd prefer another perspective here. Think incrementally: `kw patch-hub` exists and can be used. As any active project, it will always require new features, improvements and maintenance, therefore you've mapped the next steps to enhance the tool and cover developers/user needs better. ;)

davidbtadokoro replied:
For sure, I agree. This notion of "all steps needed" and "complete" isn't real. Thanks 😄

rodrigosiqueira commented:
Merged! Thanks a lot.
I'm looking forward to the other posts.
